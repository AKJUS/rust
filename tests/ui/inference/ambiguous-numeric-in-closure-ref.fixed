// Test for better error message when numeric type is ambiguous in closure parameter with reference

//@ run-rustfix

fn main() {
    let _ = (0..10).filter(|&v: &i32| v.pow(2) > 0);
    //~^ ERROR can't call method `pow` on ambiguous numeric type `{integer}`
    //~| SUGGESTION &i32
}
