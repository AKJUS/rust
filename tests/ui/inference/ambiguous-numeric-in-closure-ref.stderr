error[E0689]: can't call method `pow` on ambiguous numeric type `{integer}`
  --> $DIR/ambiguous-numeric-in-closure-ref.rs:6:35
   |
LL |     let _ = (0..10).filter(|&v| v.pow(2) > 0);
   |                              -    ^^^
   |                              |
   |                              you must specify a type for this binding
   |
help: specify the type in the closure argument list
   |
LL |     let _ = (0..10).filter(|&v: &i32| v.pow(2) > 0);
   |                               ++++++

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0689`.
